db/migration/1719842654592-AddChartConfigsTable.ts

```diff
@@ -0,0 +1,134 @@
+import { MigrationInterface, QueryRunner } from "typeorm"
+
+export class AddChartConfigsTable1719842654592 implements MigrationInterface {
+    private async createChartConfigsTable(
+        queryRunner: QueryRunner
+    ): Promise<void> {
+        await queryRunner.query(`-- sql
+            CREATE TABLE chart_configs (
+                id binary(16) NOT NULL DEFAULT (UUID_TO_BIN(UUID(), 1)) PRIMARY KEY,
+                uuid varchar(36) GENERATED ALWAYS AS (BIN_TO_UUID(id, 1)) VIRTUAL,
+                patch json NOT NULL,
+                full json NOT NULL,
+                slug varchar(255) GENERATED ALWAYS AS (JSON_UNQUOTE(JSON_EXTRACT(full, '$.slug'))) STORED,
+                createdAt datetime NOT NULL DEFAULT CURRENT_TIMESTAMP,
+                updatedAt datetime DEFAULT NULL ON UPDATE CURRENT_TIMESTAMP,
+                INDEX idx_chart_configs_slug (slug)
+            )
+        `)
+    }
+
+    private async createConfigIdColumnInChartsTable(
+        queryRunner: QueryRunner
+    ): Promise<void> {
+        // add a new `configId` column to the charts table
+        // that points to the `chart_configs` table
+        await queryRunner.query(`-- sql
+            ALTER TABLE charts
+            ADD COLUMN configId binary(16) UNIQUE AFTER type,
+            ADD CONSTRAINT charts_configId
+                FOREIGN KEY (configId)
+                REFERENCES chart_configs (id)
+                ON DELETE RESTRICT
+                ON UPDATE RESTRICT
```

Comment:
@danyx23 I'm wondering if this should be `ON DELETE CASCASE ON UPDATE CASCASE` instead. I put in the more restrictive setting to be on the safe side, but now that I'm working with it, this makes deleting charts (or indicator charts) a bit cumbersome. (The `ON UPDATE` clause doesn't really matter)

Reply:
I thought about this too when I initially reviewed it but also thought the additional safety might be useful. Are there places where you think the "cumbersominess" is particularly bad?

Reply:
No, it's okay. I was just wondering if I made my life more difficult than it has to be, but I also like the extra safety 👍🏻